packages/components/src/internal/components/ontology/OntologyTreePanel.tsx

```diff
@@ -18,14 +19,131 @@ export class OntologyPath {
 
 type PathNode = OntologyPath & TreeNode;
 
+// exported for jest testing
+export const FilterIcon = props => {
+    const { node, onClick, filters = new Map<string, PathModel>() } = props;
+
+    // TODO should we tag the parent nodes that have children with filters on them, which path if we are working from a concept code
+    // const [childSelected, setChildSelected] = useState<boolean>([...filters].some(filter => filter?.path?.startsWith(node?.data?.path)));
+    // useEffect(() => {
+    //     setChildSelected(
+    //         [...filters.values()].some(filter => {
+    //             if (!node?.data || !filter?.path) return false;
+    //
+    //             return filter.path.startsWith(node.data.path) && filter.path !== node.data.path;
+    //         })
+    //     );
+    // },[node, filters]);
+
+    const clickHandler = useCallback(
+        evt => {
+            evt.stopPropagation();
+            onClick?.(node.data);
+        },
+        [node, onClick]
+    );
+
+    return (
+        <i className={classNames('fa fa-filter', { selected: filters.has(node?.data?.code) })} onClick={clickHandler} />
+    );
+};
+
+interface OntologyTreeHeaderProps {
```

Comment:
nearly all of the props listed in this interface are duplicates from `TreeNodeProps` would it be possible to extend `TreeNodeProps` and just add in the `filters` and `onFilterClick` here?

Reply:
done.